pkg/k8s/portforward.go

```diff
@@ -73,48 +75,29 @@ func NewPortForward(
 	host string, localPort, remotePort int,
 	emitLogs bool,
 ) (*PortForward, error) {
-	timeoutSeconds := int64(30)
-	podList, err := k8sAPI.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{TimeoutSeconds: &timeoutSeconds})
-	if err != nil {
-		return nil, err
-	}
+	errStr := fmt.Errorf("no running pods found for %s", deployName)
 
-	podName := ""
-	for _, pod := range podList.Items {
-		if pod.Status.Phase == corev1.PodRunning {
-			grandparent, err := getDeploymentForPod(ctx, k8sAPI, pod)
-			if err != nil {
-				log.Warnf("Failed to get deploy for pod [%s]: %s", pod.Name, err)
-				continue
-			}
-			if grandparent == deployName {
-				podName = pod.Name
-				break
-			}
+	deploy, err := k8sAPI.AppsV1().Deployments(namespace).Get(ctx, deployName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, errStr
```

Comment:
We could be more specific with the error here:
```suggestion
			return nil, fmt.Errorf("failed to get deployment: %s", deployName)
```